lenskit-solver/src/main/java/org/lenskit/featurizer/FeatureExtractorUtilities.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2014 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+package org.lenskit.featurizer;
+
+import org.lenskit.space.IndexSpace;
+
+import java.util.List;
+
+/**
+ * Some utility functions for feature extraction.
+ *
+ * @author <a href="http://www.grouplens.org">GroupLens Research</a>
+ */
+public class FeatureExtractorUtilities {
+    private FeatureExtractorUtilities() {}
+
+    /**
+     * A utility function to construct {@link Feature} from a key-value pair. It will add the constructed feature
+     * into features unless update is false and key is not in the indexName of indexSpace.
+     */
+    static public void getOrSetIndexSpaceToFeaturize(List<Feature> features,
```

Comment:
Why isn't this just a method in `IndexSpace`, or maybe a static utility method on `Feature`?